### DISCUSSION TITLE
URI / hyperlink_rules

### DISCUSSION THREAD

**Discussion by adriancmiranda** [wezterm/wezterm#3792] [Q&A] [answered]
```lua
local wezterm = require('wezterm')

wezterm.on('open-uri', function(window, pane, uri)
	local start, match_end = uri:find('vscode://');
	if start == 1 then
		local cwd = pane:get_current_working_dir()
		local file_path = cwd .. uri:sub(match_end+1)
		local url = 'vscode://file' .. file_path
		window.open_with(url)
		return false;
	end
end)

-- @see https://wezfurlong.org/wezterm/config/lua/config/hyperlink_rules.html
-- @see https://gist.github.com/Eliot00/fbd906cc13eed021c28dbda870b7dbb8
local Hyperlinks = {} -- wezterm.default_hyperlink_rules()

-- make username/project paths clickable. this implies paths like the following are for github.
-- ( "nvim-treesitter/nvim-treesitter" | 'unknjs/unkn' | "wez/wezterm.git" )
-- as long as a full url hyperlink regex exists above this it should not match a full url to
-- github or gitlab / bitbucket (i.e. https://gitlab.com/user/project.git is still a whole clickable url)
table.insert(Hyperlinks, {
	regex = [[["']{1}([\w\d]{1}[-\w\d]+)(/){1}([-\w\d\.]+)["']{1}]],
	format = 'https://www.github.com/$1/$3',
})

-- make uri clickable
table.insert(Hyperlinks, {
	regex = [[\/(?:\\.|[^\r\n\\/ ]|\/(?=[^\r\n\\/ ]))+|~\/(?:\\.|[^\r\n\\/ ]|\/(?=[^\r\n\\/ ]))+]],
	format = 'vscode://$0'
})

-- wbthomason/packer.nvim | wez/wezterm
-- ~/Sites
-- /home/Sites

return Hyperlinks
```

```lua
local Hyperlinks = require('settings.hyperlinks')

local config = {}
if wezterm.config_builder then
	config = wezterm.config_builder()
end

config.hyperlink_rules = Hyperlinks

return config;
```

Hey guys, how's it going?
I recently downloaded wezterm and I've been enjoying it a lot, but I've been trying to set up a configuration to link local files with nvim and/or vscode, but it seems to be giving a different result than expected.

1- In this section ( "nvim-treesitter/nvim-treesitter" | 'unknjs/unkn' | wbthomason/packer.nvim | wez/wezterm | "wez/wezterm.git" ), it doesn't show the link only on /packer.nvim or /wezterm.

2- Does the open-uri event exist? (I came across this piece of code [in another issue](https://github.com/wez/wezterm/issues/3012), but it doesn't seem to have much effect. It does open vscode, but not with the file I clicked on.)

3 - How can I debug the event?

EDIT:

That's my regular expression (in javascript) returning as expected

```js
var texto = `
texto qualquer com muitas expressões
e também expressões, como as seguintes: /caminho /outrocaminho\\ partedooutrocaminho ~/mais_outro-caminho que devem fazer parte do restante da informação que iremos usar para captar nossas expressões  | wbthomason/packer.nvim | wez/wezterm 
-- ~/Sites
-- /home/Sites
`;

texto.match(/\/(?:\\.|[^\r\n\\/ ]|\/(?=[^\r\n\\/ ]))+|~\/(?:\\.|[^\r\n\\/ ]|\/(?=[^\r\n\\/ ]))+/g);
```

**Comment by wez** [comment, marked as answer]
Most likely there's an issue with your regex and the text isn't being turned into a hyperlink.  If you don't see it underlined, then it won't be clickable.

wezterm uses rust's regex engine, not javascript's regex engine, so testing there may be misleading.
In addition, lua has different rules around backslash string escapes than javascript.

I'd suggest using [ShowDebugOverlay](https://wezfurlong.org/wezterm/config/lua/keyassignment/ShowDebugOverlay.html) (default is <kbd>CTRL+SHIFT+L</kbd> to evaluate `window:effective_config().hyperlink_rules` to see if the individual regexes match what you expect, and iterate on your lua string escapes until they do.

The supported regex syntax is linked from the [hyperlink_rules](https://wezfurlong.org/wezterm/config/lua/config/hyperlink_rules.html) docs.
